### PR TITLE
🐛 Source Marketo : Remove deprecated fields for the Lead sync 🚨🚨

### PR DESCRIPTION
**Pull request by CyprienBarbault**
## What
Removed deprecated fields from the Marketo API

https://github.com/airbytehq/airbyte/issues/25064

## Recommended reading order
1. `x.java`
2. `y.python`

## 🚨 User Impact 🚨
These might be breaking changes as we are removing some fields from the sync
The following columns are being removed : 
- site
- billingStreet
- mktoCompanyNotes
- salutation
- middleName
- mobilePhone
- fax
- dateOfBirth
- personTimeZone
- doNotCallReason
- department

## Pre-merge Checklist
*Expand the relevant checklist and delete the others.*

<summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [x] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [x] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [x] Connector version has been incremented
    - [x] Version has been bumped according to our [Semantic Versioning for Connectors](https://docs.airbyte.com/contributing-to-airbyte/#semantic-versioning-for-connectors) guidelines
    - [x] `Dockerfile` has updated version
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` with an entry for the new version. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)

- [x] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub and connector version bumped by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

image

### CODE REVIEW

File: airbyte-integrations/connectors/source-marketo/Dockerfile [hunk 1]
```diff
@@ -34,5 +34,5 @@ COPY source_marketo ./source_marketo
 ENV AIRBYTE_ENTRYPOINT "python /airbyte/integration_code/main.py"
 ENTRYPOINT ["python", "/airbyte/integration_code/main.py"]
 
-LABEL io.airbyte.version=0.1.11
+LABEL io.airbyte.version=2.0.3
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-marketo/Dockerfile:37]:
```suggestion
LABEL io.airbyte.version=2.0.0
```

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
Thanks for the contribution @CyprienBarbault made some suggestions.

**Comment by CyprienBarbault** [comment]
@marcosmarxm What are the next steps here ? When will this be merged ?

**Comment by marcosmarxm** [comment]
> It is breaking our sync and blocking our stakeholders

You can build locally the new version and use a dev version during the review period.

**Comment by marcosmarxm** [comment]
Hello @CyprienBarbault these fields aren't deprecated. Check Marketo documentation note:
> * Filter type is unavailable for some subscriptions.  If unavailable for your subscription, you will receive an error when calling the Create Export Lead Job endpoint (“1035, Unsupported filter type for target subscription”).  Customers may contact Marketo Support to have this functionality enabled in their subscription.

https://developers.marketo.com/rest-api/bulk-extract/bulk-lead-extract/

My recommendation is to close the issue and open one to improve the connector to not break when this happens.

**Comment by CyprienBarbault** [comment]
Hey @marcosmarxm you were right, the failing fields had been hidden via our Marketo team. This was all due to bad communication.

Sorry for the time you spent investigating this MR. 

Note: you're also right on one point, the sync should not fail in this situation, maybe that's something to raise in an other ticket

**Pull request closed without merging**